github.go

```diff
@@ -62,11 +69,39 @@ type GitHubReleaseAsset struct {
 	Name string
 }
 
+func ParseUrlIntoGithubInstance(url string, apiv string) (GitHubInstance, *FetchError) {
+  var instance GitHubInstance
+
+  regex, regexErr := regexp.Compile("https?://(?:www\\.)?(.+?\\.com).*")
```

Comment:
Couldn't a user have GitHub Enterprise on a different TLD that isn't `.com`? Perhaps you should use [url.Parse](https://golang.org/pkg/net/url/#Parse) and extract hostname from that?